core/src/repository/datasource/DataSourceMapper.php

```diff
@@ -2,129 +2,137 @@
 
 namespace harmony\core\repository\datasource;
 
-use harmony\core\repository\BaseEntity;
-use harmony\core\repository\mapper\Mapper;
+use harmony\core\repository\mapper\GenericMapper;
 use harmony\core\repository\query\Query;
 use harmony\core\shared\collection\GenericCollection;
 
+/**
+ * @template   Tentity
+ * @template   Tdata
+ * @implements GetDataSource<Tentity>
+ * @implements PutDataSource<Tentity>
+ */
 class DataSourceMapper implements GetDataSource, PutDataSource, DeleteDataSource
 {
-    /** @var GetDataSource */
+    /** @var GetDataSource<Tdata> */
     protected $getDataSource;
-    /** @var PutDataSource */
+    /** @var PutDataSource<Tdata> */
     protected $putDataSource;
     /** @var DeleteDataSource */
     protected $deleteDataSource;
 
-    /** @var Mapper */
-    protected $toInMapper;
-    /** @var Mapper */
-    protected $toOutMapper;
+    /** @var GenericMapper<Tentity, Tdata> */
```

Comment:
Same here. We should use pascal case for the types.

Reply:
Done.